inc/azure_c_shared_utility/httpapi.h

```diff
@@ -51,7 +51,9 @@ HTTPAPI_ALLOC_FAILED,                        \
 HTTPAPI_INIT_FAILED,                         \
 HTTPAPI_INSUFFICIENT_RESPONSE_BUFFER,        \
 HTTPAPI_SET_X509_FAILURE,                    \
-HTTPAPI_SET_TIMEOUTS_FAILED                  \
+HTTPAPI_SET_TIMEOUTS_FAILED,                 \
+HTTPAPI_SET_SECURE_PROTOCOLS_FAILURE         \
```

Comment:
>HTTPAPI_SET_SECURE_PROTOCOLS_FAILURE [](start = 0, length = 36)

I took a look and these error codes haven't been modified since at least 2016.  And this is for now pretty Windows specific.

Even though we don't make same back-compat promises on c-util as we do other places in SDK, i'd feel better leaving these as is and returning "HTTPAPI_ERROR" if the SetSecure fails.